packages/odyssey-react-mui/src/Checkbox.tsx

```diff
@@ -82,7 +87,24 @@ const Checkbox = ({
   value,
 }: CheckboxProps) => {
   const { t } = useTranslation();
-  const [isCheckedValue, setIsCheckedValue] = useState(isDefaultChecked);
+  const useControlledValue = <S,>(value: S | undefined): ReturnType<typeof useState<S>> => {
+    const isControlled = useRef(value !== undefined);
+    const [stateValue, setStateValue] = useState(value);
+
+    useEffect(() => {
+      setStateValue(value);
+    }, [value])
+
+    return [
+      stateValue,
+      isControlled.current ? () => undefined : setStateValue,
```

Comment:
Potentially try using `void`:
```suggestion
      isControlled.current ? void : setStateValue,
```